module/VuFind/src/VuFind/Db/Service/UserListService.php

```diff
@@ -289,4 +329,22 @@ public function getListsContainingRecord(
         $results = $query->getResult();
         return $results;
     }
+
+    /**
+     * Helper function to append type check into a dql
+     *
+     * @param string       $dql        Dql string
+     * @param array        $parameters Parameters for dql
+     * @param string|array $types      User list types to get.
```

Comment:
Maybe we should make this consistent with other comments:

```suggestion
     * @param string|array $types      Types of user lists to get. Set to an empty array to get all.
```

(Probably also want to change `array` to `string[]` as suggested elsewhere, but that will mess up the alignment, so I didn't incorporate that into my suggestion above, since it's probably best to do all of those changes at once and then use phpcbf to fix the alignment).